src/main/java/com/zs/exercise2/Dictionary.java

```diff
@@ -0,0 +1,219 @@
+        }
+        return correct;
+    }
+
+    static void addInList(Node node, ArrayList<String>  matchWord , String word){
+        if(node.isEndOfWord ){
+            matchWord.add(word);
+        }
+
+        for(int i=0;i<ALPHABET_SIZE;i++) {
+            if(node.child[i] != null) {
+                addInList(node.child[i],matchWord, word + (char)('a' + i) );
+            }
+        }
+    }
+
+    static ArrayList<String> matchWord(String word){
+        Node node = head;
+        int len = word.length();
+        ArrayList<String> matchWord = new ArrayList<String>();
+
+        for(int i=0;i<len;i++) {
+            char ch = word.charAt(i);
+
+            if(node.child[ch-'a'] == null){
+                return matchWord;
+            }
+            node = node.child[ch - 'a'];
+        }
+        addInList(node , matchWord , word);
+        return matchWord;
+    }
+
+    public static void main(String st[]){
+        Scanner in = new Scanner(System.in);
+        head = new Node();
+
+        System.out.println("Please read the instruction carefully");
+        System.out.println("1 - word insert in dictionary");
+        System.out.println("2 - check word in dictionary");
+        System.out.println("3 - find meaning in dictionary");
+        System.out.println("4 - find similar word");
+        System.out.println("5 - word start with ");
+        System.out.println("6 - for exit ");
+
+        int choice;
+        do{
+            String word, meaning;
+            boolean check;
+
+            System.out.print("Please choose which operation you want to perform = ");
+            choice = in.nextInt();
+
+            switch(choice){
+                case 1:
+                    System.out.print("Please enter word = ");
+                    word = in.next();
+                    System.out.print("Please enter meaning = ");
+                    meaning = in.next();
+                    insert(word,meaning);
+                    break;
+
+                case 2:
+                    System.out.print("Please enter which you want to check = ");
+                    word = in.next();
+                    check = search(word);
+                    if(check) {
+                        System.out.println("Yes this word is available in dictionary");
+                    }else {
+                        System.out.println("No, this word isn't present in dictionary");
+                    }
+                    break;
+
+                case 3:
+                    System.out.print("Please enter the word = ");
+                    word = in.next();
+
+                    check = search(word);
+                    if(check){
+                        meaning = meaning(word);
+                        System.out.println(word + " : " + meaning);
+                    }else{
+                        System.out.println("Sorry,this word isn't present in dictionary");
+                    }
+                    break;
+
+                case 4:
```

Comment:
Please check this use case it will give "java.lang.ArrayIndexOutOfBoundsException"

Reply:
can you share the input on which this exception comes.